rootfs/etc/haproxy/template/haproxy.tmpl

```diff
@@ -108,6 +108,18 @@ frontend httpfront
 {{ range $server := $ing.HAServers }}
 {{ if $server.UseHTTP }}
 {{ range $location := $server.Locations }}
+    http-request set-var(req.ingress_orig_path) path
+    acl {{ $location.Backend }}_path_match var(req.ingress_orig_path) -m beg {{ $location.Path }} if or ishost-{{ $server.HostnameLabel }} ishost-{{ $server.HostnameLabel }}-alias
```

Comment:
Add `ishost-...-alias` inside a condition - ne Alias "".

Prefer loop acls+setvar in a separate loop of reqrep. Worst from the point of view of the template but better to the final cfg.